checks/ban-mutable-static/build.gradle

```diff
@@ -0,0 +1,49 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+buildscript {
+    repositories {
+        mavenCentral()
+        google() // For Gradle 4.0+
+    }
+
+    dependencies {
+        classpath 'com.android.tools.build:gradle:4.1.1'
+    }
+}
+
+plugins {
+    id "net.ltgt.errorprone" version "2.0.1" apply false
+    id 'com.github.sherter.google-java-format' version '0.9'
+}
+
+apply plugin: 'com.vanniktech.maven.publish'
```

Comment:
Do we want to make this publishable in its current form? I am pretty sure the rule only applies to specific codebases, such as other EP checkers. If we wanted this to be generally useful, it would need some sort of annotation check to enable the checker (e.g. a package level `@DisallowMutableStatic` annotation or similar).

Reply:
@nimakarimipour ? Thoughts here?

Reply:
I actually removed the publication feature from this checker as it is only useful internally for now and it found all mutable static fields (shallow cases). However, I am interested into completing this in future to cover deep cases as I find it a very useful checker honestly. Maybe we can complete this later in a follow up PRs and cover deep cases as well.

c85d32e3adbf73b23121b4a25d819b78ea4dda5b